compiler/modules/types/listable.bal

```diff
@@ -0,0 +1,148 @@
+// ListableTypes are types in which each expressible subtype can be reasonably
+// represented by a list of values. counter-eg: `int` is not a ListableType since
+// uint32 can't be reasonably represented by listing all values.
+type ListableType float|string;
+
+const LT = -1;
+const EQ = 0;
+const GT = 1;
+
+type Order GT|EQ|LT;
+
+function listUnion(ListableType[] v1, ListableType[] v2, ListableType[] result) {
```

Comment:
Accepting `result` as a pram feels bit unclean, but this avoids a cast.

Reply:
I think adding the param is the right choice, but I would add a word (maybe `into`) to the name to make it clear that this is not working functionally (unlike `int`).

Reply:
I prefixed `add`, could have gone with `collect` or `accumulate`.